pkg/probe/probe.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package probe provides k8s liveness / readiness probe.
```

Comment:
remove all mentions of k8s. this is a generic problem.

Reply:
Is the clock the same on host and in the virtual container environment?
What's the solution for clock skew -- which is a very real problem, that even has special sections dedicated to it (e.g. https://docs.turbinelabs.io/reference/quickstart.html)?

Reply:
Correct me if I'm wrong, I think the timestamp on the file and the file probing CLI are both using the wall clock inside the container. The host/VM wall clock is not used. So the clock skew between host and container env should not impact the prober time comparison.

Reply:
The timestamp is local to the container, both controller and client should be in the same container. I suspect that the time skew like mentioned in that link wouldn't be applied here.

Having said that, indeed this is very complicated and can behave incorrectly in several ways. Maybe better to be a bit more simplistic?  @myidpt wdyt?

Reply:
What about starting with this approach, and add alternatives in the future, when needed?

I agree that for less potential time issues and better performance, we could probe based on the existence of the file (or similar). That's not much implementation overhead and I think we could add that in the future.

Reply:
The timestamp is local to docker container, but the controller runs on the host machine. If the time flows at different rates, there'll be issues with this approach. That depends how many k8s environments virtualize the clock versus sharing it with the guests. It's hard to tell until we get usage, but then it might be too late to change :)

Reply:
I also got confused, but the skew between the container and the host wouldn't make a point. The file update interval is used to check the validity of the file, and it's local to the container.
The host time is used for when the checking happens, but the probe should succeed whenever it's called.

I'm afraid that my code could confuse people (including me) a lot. So I renamed the "probe interval" variables in this PR to "update interval" or something. That does not have to be same as the probe in the yaml file.